### ISSUE TITLE
Import likes from Soundcloud account

### ISSUE THREAD

**Issue by maxkraft7** [TeamNewPipe/NewPipe#3783]
#### Import likes from Soundcloud account

When I like some song on my pc it would be cool if that song shows up in my newpipe-app aswell. 

#### How will you/everyone benefit from this feature?
You don't have to maintain seperate playlists

**Comment by Stypox** [comment]
Does SoundCloud provide a way to export likes? If so then this is doable, otherwise it is not.

**Comment by wb9688** [comment]
@Stypox: See e.g. https://soundcloud.com/wb9688/likes. ^(Don't judge what I listened 4 years ago lol)

**Comment by maxkraft7** [comment]
> Does SoundCloud provide a way to export likes? If so then this is doable, otherwise it is not.

The likes are publicly available by default so you just have to scrape them from soundcloud.com/<yourname>/likes. With plain javascript you could get all the div-Tags with
`document.querySelectorAll('.sound__content')`

**Comment by dakiito** [comment]
bump

**Comment by EthoIRL** [comment]
bump

**Comment by itsokk** [comment]
this would be really useful, why is this open for 4 years now?

**Comment by picklemaster360** [comment]
I'd really like this feature as well. If someone puts up a monetary bounty for this on one of those crowd-sourced opensource dev funding platforms I'll happily contribute.

**Comment by Mathis-Aubry** [comment]
hey, we still need this in 2025 !
thank you for your work

**Comment by watermelon42** [comment]
@opusforlife2 I have a local build of NewPipe that can load "Reposts" of a Soundcloud user, it should be easy to change to import "Likes".
I had to change code in NewPipeExtractor and NewPipe. It's not gourgeous because the API url is slightly different (`/user` vs. `/stream/user`), but it works fine and it's not a huge change.

Should I make a PR for this? I saw the comment "Please do not open pull requests for new features now, only bugfix PRs will be accepted.", so maybe I should wait.

PS. I can't get the tests to pass locally, even without any changes to the repos. Not for `master`, nor for the release tags. Are you aware of this?

**Comment by absurdlylongusername** [comment]
@watermelon42 How exactly does this work? Is it just another tab in the view for a SoundCloud channel? (e.g. currently we have tracks, playlists, albums and about).

PRs for not-so-big features can be opened as far as I'm aware, and I don't think adding a new tab for likes, or the ability to get a user's likes playlist is that big as it's essentially just a playlist but a different url, so can reuse whatever logic we use for extracting soundcloud playlists (which I presume you probably did)

**Comment by absurdlylongusername** [comment]
@watermelon42 And yeah, tests are borked for a long time. I've fixed some of the extractor ones locally, but I haven't fixed the rest. You can see them failing on the CI builds

**Comment by watermelon42** [comment]
@davidasunmo Yeah, it's a tab: https://postimg.cc/TyVYTXMf. The only changes I did were in the extractor, to support the different URL and response body, and in the app to add the tab and set a boolean to get the playlist feature working.

I'll update the code to use likes instead of reposts and open a PR in the extractor first. After that is merged, I'll open a PR in the app. Is this the right process?

I have some other questions though:
- what about tests? if they are not working, should I add new ones?
- how should I handle localization of the `Likes` string? I can't even translate that in my native language (romanian) 😅

**Comment by absurdlylongusername** [comment]
@watermelon42 
> I'll update the code to use likes instead of reposts and open a PR in the extractor first. After that is merged, I'll open a PR in the app. Is this the right process?

I believe so

> what about tests? if they are not working, should I add new ones?

Please, I beg of you, **DO NOT** write code that has not been tested. The codebase is not very good or well tested as it is, so I'd much prefer all new code we add is done using best practices.

That being said, which tests are not working? The ones that are already failing on CI builds? If they are already failing then it has nothing to do with your code. But you should absolute write tests for the new stuff you make: we want to make sure that the code itself can verify it works instead of that having to be done by the user.

The extractor is a lot easier to test than the actual app, for which we don't really have much tests at all. But we will add proper testing as part of the rewrite.

Since writing tests for the app isn't straightforward, the simplest thing I'd suggest is to make test cases/scenarios that can be tested manually to ensure it works (which can be converted to automated tests later) and do these tests yourself. E.g. going to the likes tab on a channel, scrolling down to load more, loading until you have the whole list. Loading likes for a channel that doesn't have any likes, or only has one page worth, etc.

Put these test cases in your PR, and we can make an issue later on to make automated tests to do these tests.

> how should I handle localization of the Likes string? I can't even translate that in my native language (romanian) 😅

[Something something weblate](https://teamnewpipe.github.io/documentation/09_maintainers_view/#update-weblate), but I don't really know honestly. Ask in the matrix chat

**Comment by watermelon42** [comment]
I made the PR for the extractor:
https://github.com/TeamNewPipe/NewPipeExtractor/pull/1308

I'm not sure if I did the right thing w.r.t. the version of the lib, lmk if you prefer something else.

**Comment by watermelon42** [comment]
And this PR for the app:
https://github.com/TeamNewPipe/NewPipe/pull/12367
